channels/generic/websockets.py

```diff
@@ -243,33 +268,36 @@ class WebsocketMultiplexer(object):
     The opposite of the demultiplexer, to send a message though a multiplexed channel.
 
     The multiplexer object is passed as a kwargs to the consumer when the message is dispatched.
-    This pattern allows the consumer class to be independant of the stream name.
+    This pattern allows the consumer class to be independent of the stream name.
     """
 
     stream = None
     reply_channel = None
 
-    def __init__(self, stream, reply_channel):
+    def __init__(self, stream, reply_channel, json_encoder=None):
```

Comment:
Had to keep this as a constructor arg/instance var, since there's no way to propagate the demuxer's class variable into the `WebsocketMultiplexer` _class_, and the demuxer directly instantiates this class so there's no existing way to inject a custom `WebsocketMultiplexer` class.

I suppose we could make a class var in `WebsocketDemultiplexer` to hold a custom multiplexer class, but then the developer would have to remember to define custom json encoders/decoders on custom `WebsocketDemultiplexer` _and_ `WebsocketMultiplexer` subclasses, _and_ remember to hook up the custom `WebsocketMultiplexer` subclass to the custom `WebsocketDemultiplexer` class. The code here plays a little hot potato with the `json_encoder`, but it feels like it would provide an easier API in practice.